rclcpp/include/rclcpp/create_subscription.hpp

```diff
@@ -76,6 +115,34 @@ create_subscription(
   return std::dynamic_pointer_cast<SubscriptionT>(sub);
 }
 
+/// Convenience method to create a timer
+/**
+ *
+ * \tparam CallbackT callback template
+ * \param nanos period to exectute callback
+ * \param callback
+ * \param group
+ * \param node_base_
+ * \param node_timers_
+ * \return
+ */
+template<typename CallbackT>
+typename rclcpp::WallTimer<CallbackT>::SharedPtr
+create_timer(
+  std::chrono::nanoseconds nanos,
+  CallbackT callback,
+  rclcpp::callback_group::CallbackGroup::SharedPtr group,
+  rclcpp::node_interfaces::NodeBaseInterface * node_base_,
+  const rclcpp::node_interfaces::NodeTimersInterface::SharedPtr & node_timers_)
+{
+  auto timer = rclcpp::WallTimer<CallbackT>::make_shared(
+    nanos,
+    std::move(callback),
+    node_base_->get_context());
+  node_timers_->add_timer(timer, group);
+  return timer;
+}
```

Comment:
This needs to be in its own header...

Reply:
And should replace the code in `node_impl.hpp`.

Reply:
You can also just copy the content of this function into the body of create_subscription, if you don't want to deal with all of that, but if we add this function publicly we should do it properly.

Reply:
I'm happy to make a new `create_timer` header if with the same signature.

Reply:
Actually, there's already a `create_timer` header, so will use that and replace the code in `node_impl.hpp`.

Reply:
Moved the definition to `create_timer.hpp`, need to replace the code in `node_impl.hpp`.

Reply:
The new timer code is now used in `node_impl.hpp`.